Mathlib/Algebra/Group/Subgroup/Pointwise.lean

```diff
@@ -340,6 +345,20 @@ theorem conj_smul_subgroupOf {P H : Subgroup G} (hP : P ≤ H) (h : H) :
 
 end Monoid
 
+section Group
+variable [Group H] [MulAction G H] [IsScalarTower G H H]
+
+@[to_additive]
+lemma pairwiseDisjoint_smul [SetLike S H] [SubgroupClass S H] (s : S) :
+    (Set.range fun a : G ↦ a • (s : Set H)).PairwiseDisjoint id := by
```

Comment:
Can you also prove the version for right cosets?

Reply:
This looks like the kind of statement that might also be nice unfolded: if `a != b` then `Disjoint (a • s) (b • s)`. That way, it's easy to access the disjointness (eg for rewriting with lemmas which ask for a disjointness assumption) without going through Set.range and PairwiseDisjoint and Function.onFun. (Of course the version with PairwiseDisjoint should still exist!)

Reply:
> if `a != b` then `Disjoint (a • s) (b • s)`
... except that it's wrong! It's really important that we take the range before talking about disjointness

Reply:
> Can you also prove the version for right cosets?

I thought my statement already covered it, but it seems like it can't: The left version must use `smul_coe_set` while the right version must use `op_smul_coe_set`